lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -230,6 +230,72 @@ def add_merchant_description(post, options)
         merchant[:locality] = options[:merchant_descriptor_locality] if options[:merchant_descriptor_locality]
       end
 
+      def add_stored_credentials(post, payment, options)
+        stored_credential = options[:stored_credential]
+        if stored_credential
+          options = stored_credential_options(stored_credential, options.fetch(:reason_code, ''))
+          post[:processingInformation][:commerceIndicator] = options.fetch(:transaction_type, 'internet')
+          stored_credential[:initial_transaction] ? initial_transaction(post, options) : subsequent_transaction(post, options)
+        end
+      end
+
+      def stored_credential_options(options, reason_code)
+        transaction_type = options[:reason_type]
+        transaction_type = 'install' if transaction_type == 'installment'
+        initiator = options[:initiator] if  options[:initiator]
+        initiator = 'customer' if initiator == 'cardholder'
+        stored_on_file = options[:reason_type] == 'recurring'
+        options.merge({
+          transaction_type: transaction_type,
+          initiator: initiator,
+          reason_code: reason_code,
+          stored_on_file: stored_on_file
+        })
+      end
+
+      def add_processing_information(initiator, merchant_initiated_transaction_hash = {})
+        {
+          authorizationOptions: {
+            initiator: {
+              type: initiator,
+              merchantInitiatedTransaction: merchant_initiated_transaction_hash
+            }
+          }
+        }.compact
+      end
+
+      def initial_transaction(post, options)
```

Comment:
It looks like we might need to add the `credentialStoredOnFile` field to indicate that the credentials are being stored.

Reply:
For the [initial transaction ](https://developer.cybersource.com/docs/cybs/en-us/payments/developer/ctv/rest/payments/credentials-intro/credentials-maxtrix/credentials-maxtrix-initial.html)looks like this field is not necessary

Reply:
@javierpedrozaing I think @curiousepic  is right, based on guides seems to be the case that we should sen this value on the initial transaction:

![Screen Shot 2023-03-21 at 12 20 37 PM](https://user-images.githubusercontent.com/133580/226690711-ca075a81-b31b-4f55-9fbd-9491e3cd5670.png)

**Developer Notes:**

for concrete examples check the configuration tab on the API explorer, so you can be sure that examples are more relevant, for the conceptual part check de [PDF docs](https://developer.cybersource.com/content/dam/docs/cybs/en-us/payments/developer/ctv/rest/payments.pdf) they are way better than all that madness on the guides site

Reply:
It seems I was using a wrong reference, tahnks, updated